src/nimvscode/vscodeExt.nim

```diff
@@ -51,42 +51,40 @@ proc runCheck(doc:VscodeTextDocument = nil):void =
         VscodeProgressOptions{
             location:VscodeProgressLocation.window,
             cancellable:false,
-            title:"Nim: check projection..."
+            title:"Nim: check project..."
         },
         proc():Promise[seq[CheckResult]] = check(uri.fsPath, config)
     ).then(proc(errors:seq[CheckResult]) =
         diagnosticCollection.clear()
 
-        var diagnosticMap = newMap[cstring,seq[VscodeDiagnostic]]()
+        var diagnosticMap = newJsAssoc[cstring,Array[VscodeDiagnostic]]()
```

Comment:
Is the `newJsAssoc` intentional? As it will create a JS object instead of a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), which is unless there is an issue, I'd like to tend towards.

Reply:
This is the only thing I'm curious about, otherwise I'll smash the merge button.

Reply:
Also, I checked this out locally, I think there is an unused import `sequtils`.

Reply:
I see the problem, the Map port is incomplete, I'll suggest a change there.

Reply:
I changed it mostly while figuring out what's wrong and left it for consistency.

Reply:
imho ideally those wrapper could go with some more polish into the standard library